hw/ip/pwrmgr/data/pwrmgr_testplan.hjson

```diff
@@ -11,22 +11,225 @@
+            - Checks the output `pwr_rst_req.reset_cause` doesn't change for
+              a bounded amount of time.
+            - Check that the `wakeup_info` CSR flags either `fall_through` or
+              `abort` events when capture is enabled.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: reset
+      desc: '''
+            Test random reset and reset_en.
+
+            The different reset inputs can be disabled via bits in the
+            `reset_en` CSR. Resets can be triggered either in active or
+            low power state.
+
+            **Stimulus**:
+            - Sets `reset_en` randomly.
+            - Randomly choose whether to put the unit in low power mode.
+            - Set `rstreqs_i` inputs randomly.
+
+            **Checks**:
+            - Waits for fast fsm to become active reading `ctrl_cfg_regwen`
+              until it becomes 1.
+            - Checks the `reset_status` CSRs.
+            - Checks `ip_clk_en` output has a low transition.
+            - Checks the output `pwr_rst_req.reset_cause` matches a reset
+              cause.
+            - Checks the output `pwr_rst_req.rstreqs` matches the enabled
+              resets.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: escalation_reset
+      desc: '''
+            Test escalation reset.
+
+            An escalation reset cannot be disabled.
+
+            **Stimulus**:
+            - Set `esc_rst_tx_i` inputs randomly.
+
+            **Checks**:
+            - Waits for fast fsm to become active reading `ctrl_cfg_regwen`
+              until it becomes 1.
+            - Checks the `reset_status` CSRs.
+            - Checks the output `pwr_rst_req.reset_cause` matches a reset
+              cause.
+            - Checks the output `pwr_rst_req.rstreqs` is not set.
+            - Check the output `esc_rst_rx_o` indicates an alert escalation.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: reset_wakeup_race
+      desc: '''
+            Test low power entry and reset request almost coinciding.
+
+            If a low power entry and a reset occur at nearly the same time
+            the system handles them one at a time.
+
+            **Stimulus**:
+            - Trigger reset and low power entry as described for other
+              testpoints.
+            - Make sure to test them coinciding also.
+
+            **Check**:
+            - Similar tests as for the wakeup and reset testpoints, except
+              making sure they happen per the triggering order.
+              '''
+      milestone: V2
       tests: []
     }
   ]
+  covergroups: [
+    {
+      name: wakeup_cg
+      desc: '''
+            Collects coverage related to wakeup functionality.
+
+            Covergroup contains coverpoints for the `wakeups_i` input,
+            `wakeup_en`, `wakeup_info`, and `wakeup_info_capture_dis`, 
+            and suitable crosses.
```

Comment:
Also cross wakeup signal occurring during various low power entry FSM states. Likewise with reset req.

Reply:
Hmmm, I don't see the added value of this. They are sensed in specific states: if they are asserted before (and they may) they are don't cares, and they are needed in the state that senses them or the state transition won't occur. If a new transition happens after we will probably go through another reset. 

I added setting reset randomly in time as well, in the reset testpoint.

Reply:
The value of doing this would be ensure that they don't have any unexpected side-effects when they assert in don't care states, and in the states they are sensed, the correct FSM path is taken (if different from 'regular' FSM path). IIRC there are 'fallthrough' and 'abort' paths to low power entry.

Reply:
BTW in my first comment, I said "low power entry" FSM states - not ALL FSM states. Crossing with ALL FSM states may not be useful, yes.

Reply:
Well, wakeup is unrelated to low power entry, other than a wakeup (or reset) is sensed in the slow fsm in a single state, which happens after the states related to lowpower entry.

Reply:
Looked at the design - I suppose you are right.

Reply:
Also @tjaychen, please comment on whether you think this would be useful (crossing wakeups / reset_reqs occurring in different states as both fsms enter low power). I agree that they are only sensed in some states, but I was wondering if there could be a possibility of fsms going out of whack if we do this (so it probably doesn't hurt to do that anyway).

Reply:
The fast fsm can "abort" a low power entry, not via a wakeup, but due to the following two situations: the lc, otp, or flash are not idle; the processor got an interrupt so the core_sleeping goes back low. We will test this in testpoint "aborted_lowpower".

In addition, both "reset_wakeup_race" and "lowpower_wakeup_race" testpoints will target scenarios tou are concerned about. I think we are covered.